### PR TITLE
refactor: Makes checkErrorWrapper a standalone util function

### LINKED ISSUE #140: Extract checkErrorWrapper and use it in other tests
#### Reason/Context
Tests that do `try/catch` to evaluate thrown error must be wrapped in `checkErrorWrapper` to make sure tests always work, and they fail even if the test doesn't jump into the `catch`.

#### Description
- move `checkErrorWrapper` from `parse_test.js` to separate file, `utils.js` ?
- refactor `asyncapiSchemaFormatParser_test.js` and `customValidators_test.js` to use wrapper in test cases where `try/catch` is used

### PR DESCRIPTION
**Pull request by McZenith**
**Description**

- moved checkErrorWrapper to utils
- Refactor other tests using async/await to use the util function

**Related issue(s)**
Resolves #140

### CODE REVIEW

File: test/parse_test.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ const fs = require('fs');
 const path = require('path');
 const parser = require('../lib');
 const ParserError = require('../lib/errors/parser-error');
```
Comment (derberg) [inline test/parse_test.js:7]: you can remove this line, not needed anymore

File: test/parse_test.js [hunk 2]
```diff
@@ -31,28 +31,6 @@ const eolLength = EOL.length;
  * @param  {Function} fn Function that you want to test
  * @param  {Object} validationObject Error object to evaluate against the error thrown by fn()
 */
```
Comment (derberg) [inline test/parse_test.js:33]: please move above jsdoc to utils to checkErrorWrapper function too

File: test/utils.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+const chai = require('chai');
+const chaiAsPromised = require('chai-as-promised');
+const ParserError = require('../lib/errors/parser-error');
+chai.use(chaiAsPromised);
+const expect = chai.expect;
+
+
```
Comment (derberg) [inline test/utils.js:7]: please remove this extra line, I suggest running locally `npm run lint` to see all the linter errors

**Comment by derberg** [comment]
Also please remove yarn.lock from this PR

**Comment by McZenith** [comment]
Thank you @derberg, I am actually trying to make daily incremental progress towards the issue. I will be pushing daily progress until I am able to finally resolve the issue. Thanks for the guides they are really helpful!

**Comment by derberg** [comment]
@McZenith sounds good, then please convert it to a draft PR so no one bothers you and then submit it once ready

Screenshot 2020-08-06 at 17 27 23

**Comment by McZenith** [comment]
Will do that immediately, thank you.

**Pull request closed without merging**